src/libraries/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -45,17 +55,34 @@ public override bool CanWrite
             }
         }
 
+        // We're already sending data as StreamContent,
+        // so we buffer data in memory and flush it in three cases:
+        // - When GetResponse called.
+        // - When RequestStream is getting disposed.
+        // - When user calls Flush.
         public override void Flush()
         {
-            // Nothing to do.
+            ThrowIfDisposed();
+
+            if (_isBuffered && _buffer!.Length > 0)
+            {
+                _streamBuffer.Write(_buffer!.GetBuffer().AsSpan(0, (int) _buffer!.Length));
+                _buffer!.SetLength(0);
+            }
         }
 
         public override Task FlushAsync(CancellationToken cancellationToken)
         {
-            // Nothing to do.
-            return cancellationToken.IsCancellationRequested ?
-                Task.FromCanceled(cancellationToken) :
-                Task.CompletedTask;
+            ThrowIfDisposed();
+
+            if (_isBuffered && _buffer!.Length > 0)
```

Comment:
If you wanted to get rid of all these `!`, you could change `_isBuffered` to instead be something like
```c#
[MemberNotNullWhen(true, nameof(_buffer))]
private bool IsBuffered { get; set; }
```